### PR TITLE
chore(j-s): Guard Cleanup and Validations

### PR DESCRIPTION
**Pull request by gudjong**
# Guard Cleanup and Validations

[Backend type validations](https://app.asana.com/0/1199153462262248/1207300678086945/f)

## What

- Improves type declarations in backend models.
- Improves backend dto validations.
- Rmoves unnecessary case type checks in transition roles rules.

## Why

- Improved data integrity and security.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced validation for various properties across multiple DTOs and models, ensuring more accurate data handling and input validation.

- **Improvements**
  - Updated API documentation with specific types for better Swagger documentation and improved clarity.
  - Refined property types in multiple models for better data consistency and validation.

- **Bug Fixes**
  - Corrected various decorators to ensure proper data validation and type enforcement.

- **Refactor**
  - Standardized the use of `@ApiProperty` and `@ApiPropertyOptional` decorators with explicit type information across multiple models and DTOs.

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/user/dto/updateUser.dto.ts [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
 import { ApiPropertyOptional } from '@nestjs/swagger'
 
-import type { UserRole } from '@island.is/judicial-system/types'
+import { UserRole } from '@island.is/judicial-system/types'
 
 export class UpdateUserDto {
   @IsOptional()
   @IsString()
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({ type: String })
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/user/dto/updateUser.dto.ts:16]: I'm a bit confused, why is this an optional property, but still has a `!`?

**Review by oddsson — CHANGES_REQUESTED** [review]
Just a couple of comments..

File: apps/judicial-system/backend/src/app/modules/user/dto/updateUser.dto.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/user/dto/updateUser.dto.ts:16]: Good catch, all the fields should be optional.

**Review by oddsson — APPROVED** [review]

File: apps/judicial-system/backend/src/app/modules/file/dto/updateFile.dto.ts [hunk 2]
```diff
@@ -13,40 +15,42 @@ import {
...
   @Min(0)
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({ type: Number })
   readonly orderWithinChapter?: number
 
   @IsOptional()
-  @IsString()
-  @ApiPropertyOptional()
+  @Transform(({ value }) => new Date(value))
+  @IsDate()
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/file/dto/updateFile.dto.ts:44]: Will change

**Pull request merged**